kura/org.eclipse.kura.driver.eddystone.provider/src/main/java/org/eclipse/kura/internal/driver/eddystone/EddystoneDriver.java

```diff
@@ -0,0 +1,319 @@
+        if (bluetoothLeAdapter != null) {
+            if (!bluetoothLeAdapter.isPowered()) {
+                bluetoothLeAdapter.setPowered(true);
+            }
+            this.bluetoothLeEddystoneScanner = this.bluetoothLeEddystoneService.newBeaconScanner(bluetoothLeAdapter);
+            this.bluetoothLeEddystoneScanner.addBeaconListener(this);
+            this.worker = Executors.newSingleThreadScheduledExecutor();
+            // Setup a task that monitor the scan every 30 seconds
+            this.handle = this.worker.scheduleAtFixedRate(this::monitor, 0, MONITOR_TIMEOUT, TimeUnit.SECONDS);
+        } else {
+            logger.warn("No Bluetooth adapter found ...");
+        }
+    }
+
+    private void monitor() {
+        if (!this.bluetoothLeEddystoneScanner.isScanning()) {
+            try {
+                // Perform a scan for 10 minutes
+                this.bluetoothLeEddystoneScanner.startBeaconScan(SCAN_TIMEOUT * 1000L);
+            } catch (KuraException e) {
+                logger.error("iBeacon scanning failed", e);
+            }
+        }
+    }
+
+    @Override
+    public void connect() throws ConnectionException {
+        // Not needed
+    }
+
+    @Override
+    public void disconnect() throws ConnectionException {
+        // Not needed
+    }
+
+    private void extractProperties(final Map<String, Object> properties) {
+        requireNonNull(properties, "Properties cannot be null");
+        this.options = new EddystoneOptions(properties);
+    }
+
+    @Override
+    public ChannelDescriptor getChannelDescriptor() {
+        return new EddystoneChannelDescriptor();
+    }
+
+    @Override
+    public void read(final List<ChannelRecord> records) throws ConnectionException {
+        logger.warn("Read operation not supported");
+    }
+
+    @Override
+    public void registerChannelListener(final Map<String, Object> channelConfig, final ChannelListener listener)
+            throws ConnectionException {
+
+        String channelName = (String) channelConfig.get("+name");
+        EddystoneFrameType frameType = EddystoneChannelDescriptor.getEddystoneType(channelConfig);
+        EddystoneListener eddystoneListener = new EddystoneListener(channelName, listener, frameType);
+
+        this.eddystoneListeners.add(eddystoneListener);
+    }
+
+    @Override
+    public void unregisterChannelListener(final ChannelListener listener) throws ConnectionException {
+        Iterator<EddystoneListener> iterator = this.eddystoneListeners.iterator();
+        while (iterator.hasNext()) {
+            EddystoneListener sensorListener = iterator.next();
+            if (sensorListener.getListener().equals(listener)) {
+                iterator.remove();
+            }
+        }
+    }
+
+    @Override
+    public void write(final List<ChannelRecord> records) throws ConnectionException {
+        logger.warn("Write operation not supported");
```

Comment:
According to contract, it should throw `KuraRuntimeException` with `KuraErrorCode#OPERATION_NOT_SUPPORTED`